packages/process/src/node/process.ts

```diff
@@ -28,12 +28,34 @@ export enum ProcessType {
     'Terminal'
 }
 
+/**
+ * Options to spawn a new process (`spawn`).
+ *
+ * For more information please refer to the spawn function of Node's
+ * child_process module:
+ *
+ *   https://nodejs.org/api/child_process.html#child_process_child_process_spawn_command_args_options
+ */
 export interface ProcessOptions {
     readonly command: string,
     args?: string[],
     options?: object
 }
 
+/**
+ * Options to fork a new process using the current Node interpeter (`fork`).
+ *
+ * For more information please refer to the fork function of Node's
+ * child_process module:
+ *
+ *   https://nodejs.org/api/child_process.html#child_process_child_process_fork_modulepath_args_options
+ */
+export interface ForkOptions {
+    readonly modulePath: string,
```

Comment:
+1 for having a moduleUri (as a `string`) and using `FileUri.path(moduleUri)` when calling [`fork`](https://github.com/theia-ide/theia/pull/3637/files#diff-cb285c4de3864422b5b72f7f5ea420c7R95).

Reply:
Can you explain what would be the difference in practice, and why it would be preferable?

Reply:
In general, we are not using file-system paths strings in the APIs. The URI is the preferred way. For instance, here https://github.com/theia-ide/theia/blob/cff291acebb5d987df868127f4d49677cd41ec33/packages/search-in-workspace/src/common/search-in-workspace-interface.ts#L119

or here
https://github.com/theia-ide/theia/blob/cff291acebb5d987df868127f4d49677cd41ec33/packages/file-search/src/common/file-search-service.ts#L37

Yes, I am aware, everything is on the backend.

Reply:
@kittaakos I wasn't aware your `+1` was a request, sorry.
Personally I'd prefer the types were kept inline with the documented node types.

Reply:
It was not a "real" request, just something to discuss. If the others are happy with your changes, I won't hold back the PR.